azurerm/resource_arm_storage_account.go

```diff
@@ -376,60 +466,80 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
+		d.Set("account_replication_type", strings.Split(fmt.Sprintf("%v", sku.Name), "_")[1])
 	}
 
-	if resp.AccountProperties.PrimaryEndpoints != nil {
-		d.Set("primary_blob_endpoint", resp.AccountProperties.PrimaryEndpoints.Blob)
-		d.Set("primary_queue_endpoint", resp.AccountProperties.PrimaryEndpoints.Queue)
-		d.Set("primary_table_endpoint", resp.AccountProperties.PrimaryEndpoints.Table)
-		d.Set("primary_file_endpoint", resp.AccountProperties.PrimaryEndpoints.File)
-
-		pscs := fmt.Sprintf("DefaultEndpointsProtocol=https;BlobEndpoint=%s;AccountName=%s;AccountKey=%s",
-			*resp.AccountProperties.PrimaryEndpoints.Blob, *resp.Name, *accessKeys[0].Value)
-		d.Set("primary_blob_connection_string", pscs)
-	}
+	if props := resp.AccountProperties; props != nil {
+		d.Set("access_tier", props.AccessTier)
+		d.Set("enable_https_traffic_only", props.EnableHTTPSTrafficOnly)
 
-	if resp.AccountProperties.SecondaryEndpoints != nil {
-		if resp.AccountProperties.SecondaryEndpoints.Blob != nil {
-			d.Set("secondary_blob_endpoint", resp.AccountProperties.SecondaryEndpoints.Blob)
-			sscs := fmt.Sprintf("DefaultEndpointsProtocol=https;BlobEndpoint=%s;AccountName=%s;AccountKey=%s",
-				*resp.AccountProperties.SecondaryEndpoints.Blob, *resp.Name, *accessKeys[1].Value)
-			d.Set("secondary_blob_connection_string", sscs)
-		} else {
-			d.Set("secondary_blob_endpoint", "")
-			d.Set("secondary_blob_connection_string", "")
+		if customDomain := props.CustomDomain; customDomain != nil {
+			if err := d.Set("custom_domain", flattenStorageAccountCustomDomain(customDomain)); err != nil {
+				return fmt.Errorf("Error flattening `custom_domain`: %+v", err)
+			}
 		}
-		if resp.AccountProperties.SecondaryEndpoints.Queue != nil {
-			d.Set("secondary_queue_endpoint", resp.AccountProperties.SecondaryEndpoints.Queue)
-		} else {
-			d.Set("secondary_queue_endpoint", "")
+
+		if encryption := props.Encryption; encryption != nil {
+			if services := encryption.Services; services != nil {
+				if blob := services.Blob; blob != nil {
+					d.Set("enable_blob_encryption", blob.Enabled)
+				}
+				if file := services.File; file != nil {
+					d.Set("enable_file_encryption", file.Enabled)
+				}
+			}
 		}
-		if resp.AccountProperties.SecondaryEndpoints.Table != nil {
-			d.Set("secondary_table_endpoint", resp.AccountProperties.SecondaryEndpoints.Table)
-		} else {
-			d.Set("secondary_table_endpoint", "")
+
+		// Computed
+		d.Set("primary_location", props.PrimaryLocation)
+		d.Set("secondary_location", props.SecondaryLocation)
+
+		if endpoints := props.PrimaryEndpoints; endpoints != nil {
+			d.Set("primary_blob_endpoint", endpoints.Blob)
+			d.Set("primary_queue_endpoint", endpoints.Queue)
+			d.Set("primary_table_endpoint", endpoints.Table)
+			d.Set("primary_file_endpoint", endpoints.File)
+
+			pscs := fmt.Sprintf("DefaultEndpointsProtocol=https;BlobEndpoint=%s;AccountName=%s;AccountKey=%s",
+				*endpoints.Blob, *resp.Name, *accessKeys[0].Value)
```

Comment:
Just want to confirm that we are guaranteed to have 2 keys

Reply:
Yep, afaik